src/OmniSharp.LanguageServerProtocol/Eventing/LanguageServerEventEmitter.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using OmniSharp.Eventing;
+using OmniSharp.Extensions.LanguageServer;
+using OmniSharp.Extensions.LanguageServer.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Models.Diagnostics;
+using OmniSharp.Models.Events;
+using OmniSharp.Stdio.Protocol;
+using OmniSharp.Stdio.Services;
+
+namespace OmniSharp.LanguageServerProtocol.Eventing
+{
+    public class LanguageServerEventEmitter : IEventEmitter
+    {
+        private readonly LanguageServer _server;
+
+        public LanguageServerEventEmitter(LanguageServer server)
+        {
+            _server = server;
+        }
+
+        public void Emit(string kind, object args)
+        {
+            switch (kind)
+            {
+                case EventTypes.Diagnostic:
+                    if (args is DiagnosticMessage message)
+                    {
+                        var groups = message.Results
+                            // TODO: Make this is correct
```

Comment:
what a funny comment! I don't know whether to ask you to correct the typo, or to correct the thing that needs correction. :smile: